File: Go/ExampleTests/production/catalog.go
package domain

const (
	CatalogMinimum = 0
	CatalogMaximum = 10000000
)

type Catalog struct {
	items []*CatalogItem
}

func NewCatalog() *Catalog { return &Catalog{} }

func (c *Catalog) Add(item *CatalogItem) {
	c.items = append(c.items, item)
}

func (c *Catalog) Delete(item *CatalogItem) bool {
	for i, v := range c.items {
		if v == item {
			c.items = append(c.items[:i], c.items[i+1:]...)
			return true
		}
	}
	return false
}

func (c *Catalog) Read() []*CatalogItem {
	result := make([]*CatalogItem, len(c.items))
	copy(result, c.items)
	return result
}

func (c *Catalog) Update(oldItem, newItem *CatalogItem) bool {
	for i, v := range c.items {
		if v == oldItem {
			c.items[i] = newItem
			return true
		}
	}
	return false
}

func (c *Catalog) Size() int { return len(c.items) }

File: Swift/ExampleTests/production/Catalog.swift
public let catalogMinimum = 0
public let catalogMaximum = 10000000

public class Catalog {
    private var items: [CatalogItem] = []

    public init() {}

    public func add(_ item: CatalogItem) {
        items.append(item)
    }

    public func delete(_ item: CatalogItem) -> Bool where CatalogItem: Equatable {
        if let idx = items.firstIndex(where: { $0 == item }) {
            items.remove(at: idx)
            return true
        }
        return false
    }

    public func read() -> [CatalogItem] {
        return items
    }

    public func update(_ oldItem: CatalogItem, with newItem: CatalogItem) -> Bool where CatalogItem: Equatable {
        if let idx = items.firstIndex(where: { $0 == oldItem }) {
            items[idx] = newItem
            return true
        }
        return false
    }

    public var size: Int { return items.count }
}
